### DISCUSSION TITLE
How to embed a Reddit post in my blog post?

### DISCUSSION THREAD

**Discussion by gaelgoth** [importantimport/urara#62] [Q&A] [answered]
Hello!

I would like to integrate a reddit post into my article. Reddit is not (yet) supported by sveltekit-embed. I tried to do it this way in my `+page.svelte.md` file:

```markdown
Reddit post:
<html>
<blockquote class="reddit-embed-bq" style="height:500px" data-embed-theme="dark" data-embed-height="500">      <a href="https://www.reddit.com/r/unixporn/comments/1322ms3/xragnar_multiple_deskop_support_nord_theme/">[xragnar] Multiple Deskop Support &amp; Nord theme</a><br> by      <a href="https://www.reddit.com/user/cococry">u/cococry</a> in      <a href="https://www.reddit.com/r/unixporn/">unixporn</a>    </blockquote><script async="" src="https://embed.reddit.com/widgets.js" charset="UTF-8"></script>
</html>
```

Is there a better (or cleaner) to do it?

Here is the result: 
![image](https://user-images.githubusercontent.com/55390114/235323793-d38dcdfe-5f19-4aba-a92b-4e8dd698b164.png)

**Comment by Interstellar750** [comment, marked as answer]
> My English is not good, I used a translator in my answer

Hello, this is Hubert, a rookie. 🥲

I took a look at the post you want to embed and tried the embedding method provided by Reddit, which actually provides a more convenient `<iframe>` embedding method that can be found in the source code of the [Reddit Embed](https://publish.reddit.com/embed?url=https://www.reddit.com/r/unixporn/comments/1322ms3/xragnar_multiple_deskop_support_nord_theme/) page, in a format similar to the following (with formatting):

```
<iframe
    src="https://embed.reddit.com/r/unixporn/comments/1322ms3/xragnar_multiple_deskop_support_nord_theme/?
        embed=true&
        ref_source=embed&
        ref=share&
        utm_medium=widgets&
        utm_source=embedv2&
        utm_term=23&
        theme=dark&
        showedits=false&
        created=2023-04-30T02:01:37.264Z&
        showmedia=false&
        utm_name=post_embed"
    width="640" 
    scrolling="no" 
    allowfullscreen="true" 
    sandbox="allow-scripts 
        allow-same-origin 
        allow-popups"
    style="border: none; 
        max-width: 100%; 
        border-radius: 8px; 
        display: block; 
        margin: 0px auto;" 
    height="240">
</iframe>
```

The `<iframe>` code above is the output with **Hide post content**, **Dark mode** and **Hide If Updateds** checked, the difference between them is the additional parameter in `src`:

**Hide post content:** `showmedia=false&`

**Dark mode:** `theme=dark&`

**Hide If Updateds**: 

```
showedits=false&
created=2023-04-30T02:01:37.264Z&
```

For dimensions, the height defaults to `height="500"`, which may change to `height="240"` if you have **Hide post content** enabled or if there is not enough content in the post. And the height, you can set it to `width="100%"` to fit the border:


![Screenshot 2023-04-30 11 04 37](https://user-images.githubusercontent.com/38517305/235333516-938cb4c2-7780-4c6b-a6c0-f2057a3022df.png)

You can use the above to embed it into your post. Maybe I can create a component for this later.

Hope this helps. 😆

**Reply by gaelgoth** [reply]
Thanks for your help, Hubert! I'll use an `iframe` in the meantime and open a discussion about Reddit on the `sveltekit-embed` repo.

**Comment by kwaa** [comment]
I recommend making a feature request to `sveltekit-embed` to add a Reddit component.
Until then, you can use the iframe as @Interstellar750 does
